src/client/sandbox/cookie/index.js

```diff
@@ -26,78 +28,43 @@ export default class CookieSandbox extends SandboxBase {
-
-        cookieUtils.del(document, parsedCookieCopy);
-
-        if (processedByBrowserCookieStr) {
-            // NOTE: We need to remove the '=' char if the key is empty
-            const startCookiePos = parsedCookie.key === '' ? uniquePrefix.length + 1 : uniquePrefix.length;
-
-            return processedByBrowserCookieStr.substr(startCookiePos);
-        }
-
-        return null;
-    }
-
-    static _isMatchDomain (currentDomain, cookieDomain) {
-        currentDomain = currentDomain.toLowerCase();
-        cookieDomain  = cookieDomain.toLowerCase();
-
-        if (currentDomain === cookieDomain)
-            return true;
-
-        const cookieDomainIdx = currentDomain.indexOf(cookieDomain);
-
-        return cookieDomainIdx > 0 &&
-               currentDomain.length === cookieDomain.length + cookieDomainIdx &&
-               currentDomain.charAt(cookieDomainIdx - 1) === '.';
-    }
-
     // NOTE: Perform validations that can't be processed by a browser due to proxying.
     static _isValidCookie (parsedCookie) {
         if (!parsedCookie)
             return false;
 
         // NOTE: HttpOnly cookies can't be accessed from the client code.
-        if (parsedCookie.httponly)
+        if (parsedCookie.httpOnly)
             return false;
 
         const parsedDestLocation = destLocation.getParsed();
-
-        // eslint-disable-next-line no-restricted-properties
-        const destProtocol = parsedDestLocation.protocol;
+        const destProtocol       = parsedDestLocation.protocol; // eslint-disable-line no-restricted-properties
 
         // NOTE: Hammerhead tunnels HTTPS requests via HTTP, so we need to validate the Secure attribute manually.
         if (parsedCookie.secure && destProtocol !== 'https:')
             return false;
 
+        // eslint-disable-next-line no-restricted-properties
+        if (parsedCookie.path && !cookieUtils.pathMatch(parsedDestLocation.pathname, parsedCookie.path))
+            return false;
 
         // NOTE: All Hammerhad sessions have the same domain, so we need to validate the Domain attribute manually
         // according to a test url.
         // eslint-disable-next-line no-restricted-properties
-        return !parsedCookie.domain || CookieSandbox._isMatchDomain(parsedDestLocation.hostname, parsedCookie.domain);
+        return !parsedCookie.domain || cookieUtils.domainMatch(parsedDestLocation.hostname, parsedCookie.domain);
+    }
+
+    _canSetCookie () {
+        const clientCookie = `key${nativeMethods.random.call(Math)}=value`;
+
+        nativeMethods.documentCookieSetter.call(this.document, clientCookie);
+
+        const documentCookieIsEmpty = !nativeMethods.documentCookieGetter.call(this.document);
+
+        if (!documentCookieIsEmpty)
+            nativeMethods.documentCookieSetter.call(this.document, `${clientCookie};expires=${MIN_DATE_VALUE}`);
+
+        return documentCookieIsEmpty;
```

Comment:
As I understand you can not set cookie if the `documentCookieIsEmpty` is `true`? Where am I wrong?